CONTRIBUTING.md

```diff
@@ -0,0 +1,248 @@
+
+   ```bash
+   $ tox --devenv .venv
+   $ source .venv/bin/activate
+   ```
+
+4. Create a branch for local development:
+
+   ```bash
+   $ git checkout -b name-of-your-bugfix-or-feature
+   ```
+
+   Now you can make your changes locally. Make sure you follow the project code style documented in [CODING_GUIDELINES.md](CODING_GUIDELINES.md).
+
+5. When you're done making changes, check that your code comply with the project code style and other quality assurance (QA) practices using `pre-commit`, and that unit and regression tests pass for all supported Python versions using `tox`:
+
+   ```bash
+   $ pre-commit run
+   $ tox
+   ```
+
+   Read [Testing](#testing) section below for further details.
+
+6. Commit your changes and push your branch to GitHub:
+
+   ```bash
+    $ git add .
+    $ git commit -m "Your detailed description of your changes."
+    $ git push origin name-of-your-bugfix-or-feature
+   ```
+
+7. Submit a pull request (PR) through the [GitHub](https://github.com/gridtools/gt4py) website.
+
+
+## Testing
+
+### Quality Assurance
+
+We use [pre-commit](https://pre-commit.com/) to run several auto-formatting and linting tools. You should always execute it locally before opening a pull request. `pre-commit` can be installed as a _git hook_ to automatically check the staged changes before commiting:
+
+```bash
+# Install pre-commit as a git hook and initialized all the configured tools
+pre-commit install --install-hooks
+```
+
+Or it can be executed on demand from the command line:
+
+```bash
+# Check only the staged changes
+pre-commit run
+
+# Check all the files in the repository (-a / --all-files)
+pre-commit run -a
+
+# Run only some of the tools (e.g. flake8)
+pre-commit run -a flake8
+```
+
+### Unit and Regression Tests
+
+In the GT4Py project we use the [pytest](https://pytest.org/) framework for testing our code. `pytest` comes with a very convenient CLI tool to run tests. For example:
+
+```bash
+# Run tests inside `path/to/test/folder`
+pytest path/to/test/folder
+
+# Run tests stopping immediately on first error: -x / --exitfirst
+pytest -x tests/
+
+# Run tests matching the pattern: -k pattern (supports boolean operators)
+pytest -k pattern tests/
+
+# Run tests in parallel: -n NUM_OF_PROCS (or `auto`)
+pytest -n auto tests/
+
+# Run only tests that failed last time: --lf / --last-failed
+pytest --lf tests/
+
+# Run all the tests starting with the tests that failed last time:
+# --ff / --failed-first
+pytest --ff tests/
+
+# Run tests with more informative output:
+#   -v / --verbose          - increase verbosity
+#   -l / --showlocalsflag   - show locals in tracebacks
+#   -s                      - show tests outputs to stdout
+pytest -v -l -s tests/
+```
+
+Check `pytest` documentation (`pytest --help`) for all the options to select and execute tests.
+
+To run the complete test suite we also recommended to use `tox`:
```

Comment:
Could you explain why it's better to use tox than your IDEs test running features?